frontend/src/components/review/StatusBadge.tsx

```diff
@@ -77,20 +104,30 @@ const StatusBadge = ({
         <MenuItem
           icon={<Icon as={MdCheck} height={6} width={6} />}
           onClick={async () => {
-            handleStatusChange("APPROVED");
+            checkForLastApprovedLine("APPROVED");
           }}
         >
           Approve
         </MenuItem>
         <MenuItem
           icon={<Icon as={MdTimer} height={6} width={6} />}
           onClick={async () => {
-            handleStatusChange("DEFAULT");
+            checkForLastApprovedLine("DEFAULT");
```

Comment:
It's not clear that `checkForLastApprovedLine` is going to handle the status change. Maybe call `checkForLastApprovedLine` inside `handleStatusChange` or rename one of the functions?